src/cpu/aarch64/jit_sve_512_x8s8s32x_conv_kernel.cpp

```diff
@@ -0,0 +1,1642 @@
+/*******************************************************************************
+* Copyright 2016-2020 Intel Corporation
+* Copyright 2020 FUJITSU LIMITED
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "common/c_types_map.hpp"
+#include "common/memory.hpp"
+#include "common/memory_tracking.hpp"
+#include "common/nstl.hpp"
+#include "common/type_helpers.hpp"
+#include "common/utils.hpp"
+
+#include "cpu/aarch64/jit_sve_512_x8s8s32x_conv_kernel.hpp"
+
+#define GET_OFF(field) static_cast<int32_t>(offsetof(jit_conv_call_s, field))
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace aarch64 {
+
+using namespace dnnl::impl::memory_tracking::names;
+using namespace dnnl::impl::utils;
+using namespace dnnl::impl::data_type;
+
+namespace {
+void pick_loop_order(jit_conv_conf_t &jcp, int nthr) {
+    jcp.loop_order = loop_cwgn;
+    if (jcp.ngroups > 1) {
+        jcp.loop_order = loop_ngcw;
+        if (jcp.mb < nthr)
+            jcp.loop_order = jcp.ndims == 3 ? loop_nwcg : loop_nhwcg;
+    }
+}
+} // namespace
+
+template <typename Vmm>
+void _jit_sve_512_x8s8s32x_fwd_kernel<Vmm>::prepare_output(int ur_w) {
+    int nb_oc_block
+            = jcp.is_depthwise ? jcp.nb_ch_blocking : jcp.nb_oc_blocking;
+    for (int k = 0; k < nb_oc_block; k++)
+        for (int j = 0; j < ur_w; j++) {
+            auto vmm = vmm_out(j, k);
+            eor(vmm.d, vmm.d, vmm.d);
+        }
+    if (!jcp.signed_input) {
+        eor(reg_scratch, reg_scratch, reg_scratch);
+        if (jcp.is_depthwise && !jcp.is_fast_depthwise) {
+            mov_imm(WReg(reg_tmp0_imm.getIdx()), 128);
+            dup(vmm_shift.s, WReg(reg_tmp0_imm.getIdx()));
+        } else {
+            dup(vmm_shift.b, -128);
+        }
+    }
+}
+
+template <typename Vmm>
+void _jit_sve_512_x8s8s32x_fwd_kernel<Vmm>::cvt2ps(data_type_t type_in,
+        const ZReg vmm_in, const XReg reg_base, const int offset,
+        bool mask_flag) {
+
+    auto vmm = vmm_in;
+    auto reg_addr = get_comp_addr_reg(reg_base, offset);
+    switch (type_in) {
+        // case data_type::f32:
+        // case data_type::s32: vmovups(vmm, addr); break;
+        // case data_type::s8: vpmovsxbd(vmm, addr); break;
+        // case data_type::u8: vpmovzxbd(vmm, addr); break;
```

Comment:
^^^ above comments are irrelevant [as I see these are instructions for `x64` arch]

Reply:
Thx! I'll delete it.